index.js

```diff
@@ -20,7 +20,14 @@ function toPathMatch(pattern) {
     const reg = pathToRegexp(pattern, [], { end: false });
     return ctx => reg.test(ctx.path);
   }
-  if (pattern instanceof RegExp) return ctx => pattern.test(ctx.path);
+  if (pattern instanceof RegExp) {
+    return ctx => {
+      // fix: reset lastIndex when pattern used the "g" flag.
+      // Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex
+      pattern.global && (pattern.lastIndex = 0);
```

Comment:
actually you shouldn't pass /g/ here because it is useless

Reply:
the better way to avoid this is change `pattern.test(ctx.path)` to `ctx.path.match(pattern)`

Reply:
Thanks, I have updated it.